src/components/TodoList.jsx

```diff
@@ -1,66 +1,206 @@
     constructor(props) {
         super(props);
-
         this.state = {
+            // Array of todo elements as objects { name of task, 
+            // date when task added, id of task, color of task, 
+            // text decoration style of task }
             tasks: [
-                "do one thing",
-                "another one",
-                "last one to do"
+                {
+                    taskName: "do one thing", date: new Date(),
+                    id: 0, color: "", textDecoration: ""
+                },
+                {
+                    taskName: "another one", date: new Date(),
+                    id: 1, color: "", textDecoration: ""
+                },
+                {
+                    taskName: "last one to do", date: new Date(),
+                    id: 2, color: "", textDecoration: ""
+                }
             ],
-            input: "nichego net"
+            // Main input field text value
+            input: "",
+            // Current soring option to sort tasks 
+            sortingOption: "Sort by",
+            // Id of task in edit mode
+            taskInEditId: -1,
+            // Id of last created task
+            lastCreatedId: 2
         };
-
         this.addTask = this.addTask.bind(this);
         this.removeTask = this.removeTask.bind(this);
         this.onInputChange = this.onInputChange.bind(this);
+        this.removeAllTasks = this.removeAllTasks.bind(this);
+        this.sortTasks = this.sortTasks.bind(this);
+        this.setTaskInEditId = this.setTaskInEditId.bind(this);
+        this.setNewTaskText = this.setNewTaskText.bind(this);
+        this.setTaskColor = this.setTaskColor.bind(this);
+        this.setTaskTextDecoration = this.setTaskTextDecoration.bind(this)
+    }
+    // Set color of task
+    setTaskColor(taskId, bIsImportant) {
+        const { tasks } = this.state
+        for (const task of tasks) {
+            if (task.id === taskId)
+                bIsImportant ? task.color = "red"
+                    : task.color = "black"
+        }
+        this.setState({
+            ...this.state
+        })
+    }
+    // Set text deoration style of task
+    setTaskTextDecoration(taskId, bIsDone) {
+        const { tasks } = this.state
+        for (const task of tasks) {
+            if (task.id === taskId)
+                bIsDone ? task.textDecoration = "line-through"
+                    : task.textDecoration = ""
+        }
+        this.setState({
+            ...this.state
+        })
     }
 
-    removeTask(text) {
+    // Set text of task in edit mode 
+    setTaskInEditId(todoId) {
+        this.setState({
+            ...this.state,
+            taskInEditId: todoId
+        })
+    }
+    // Change name of task
+    setNewTaskText(newText) {
+        const { tasks, taskInEditId } = this.state
+        tasks[taskInEditId].taskName = newText
+        this.setState({
+            ...this.state,
+            tasks,
```

Comment:
Same problem as above: you took a reference to an array and changed its element. your setState() doesn't change the array, since it's the same array as referenced above.

Again, read thru this
https://stackoverflow.com/questions/29537299/react-how-to-update-state-item1-in-state-using-setstate